src/systems/odometry_publisher/OdometryPublisher.cc

```diff
@@ -142,6 +145,28 @@ void OdometryPublisher::Configure(const Entity &_entity,
     this->dataPtr->odomFrame = _sdf->Get<std::string>("odom_frame");
   }
 
+  if (!_sdf->HasElement("xyz_offset"))
```

Comment:
I think you can get rid of this branch if you initialize `offset` to {0, 0, 0, 0, 0, 0}`

Reply:
Done